### DISCUSSION TITLE
Mockup GNOME-desktop: One big horizontal workspace

### DISCUSSION THREAD

**Discussion by Verhoeckx** [paperwm/PaperWM#780] [Ideas]
Can PaperWM made to be working like this:

https://penpot.jeroenverhoeckx.com/#/view/79947d4f-b8ab-816a-8003-e084136116c0?page-id=79947d4f-b8ab-816a-8003-e08413617b2e&section=interactions&index=0&share-id=79947d4f-b8ab-816a-8003-e084c0f545b9

![One big horizontal workspace](https://github.com/paperwm/PaperWM/assets/64807887/b0380e0f-c0e7-4da5-bfda-7dbaba3943cb)

The windows should _not_ automatic tile as in PaperWM (can this be turned off?).

**Comment by jtaala** [comment]
> Can PaperWM made to be working like this:

No (well, not without someone modifying it heavily).

> The windows should not automatic tile as in PaperWM (can this be turned off?).

Not quite, the closest you'll get is setting all windows (by default) to be added to the scratch layer:

![image](https://github.com/paperwm/PaperWM/assets/30424662/4680c0c2-1240-49f4-85b5-ea90f3328523)

**Reply by Verhoeckx** [reply]
Hello @jtaala,

Thank you for answering the questions so quickly!
Too bad it can't be made to work like in the mock-up. Hoped it could. But no problem of course!

Good luck with the development of PaperWM :-)!

**Reply by Lythenas** [reply]
Just to make sure there is no misunderstanding. In your mockup you mention that "All windows are placed next to each other". Isn't that the same as tiling? At least that would be my definition of it.

I think if you limit Gnomes workspaces to a 1 in the settings, PaperWM kind of implements your proposal (at least for a single monitor). Except our activity overview is somewhat broken. We also add a DnD handler for the windows and some other features. But you kind of want those if you want your windows to stay "next to each other". All other differences are probably more minor, e.g. turning off our custom window borders (through CSS) or changing where the window switcher is shown (probably needs code changes).

Note that a single workspace across all monitors is currently not possible in PaperWM, see e.g. #753.

**Comment by Verhoeckx** [comment]
> Just to make sure there is no misunderstanding. In your mockup you mention that "All windows are placed next to each other". Isn't that the same as tiling? At least that would be my definition of it.

Good question! I'm not sure what the 'official' definition is of tiling, but I see the following difference between tiling window managers and my mock-up: in most tiling window managers the windows are tilled in a rigid grid, while in my mock-up they are placed on a new screen and can move freely on that this new screen (a screen in my design corresponds with a workspace in the current GNOME-design, but is visually part of the one workspace).

> I think if you limit Gnomes workspaces to a 1 in the settings, PaperWM kind of implements your proposal (at least for a single monitor). Except our activity overview is somewhat broken. We also add a DnD handler for the windows and some other features. But you kind of want those if you want your windows to stay "next to each other". All other differences are probably more minor, e.g. turning off our custom window borders (through CSS) or changing where the window switcher is shown (probably needs code changes).

I tried this and yes, the opening of new windows is exactly (or almost) as how I would like it to behave!

Thanks for thinking along!
